data_iterator.go

```diff
@@ -185,22 +190,28 @@ func (d *DataIterator) Run() {
 			defer wg.Done()
 
 			for {
-				table, ok := <-tablesQueue
+				var table *schema.Table
+				var ok bool
+				select {
+				case table, ok = <-tablesQueue:
+				case <-ctx.Done():
+					logrus.WithField("tag", "table_iterator").Info("cancellation received, terminating goroutine abruptly")
+				}
```

Comment:
We don't need this here. The producer of the `tablesQueue` channel can close the channel and terminate this loop.